subprojects/core/src/main/java/org/gradle/execution/taskgraph/TaskInfo.java

```diff
@@ -130,11 +129,18 @@ public Throwable getTaskFailure() {
     }
 
     public boolean allDependenciesComplete() {
-        for (TaskInfo dependency : Iterables.concat(mustSuccessors, dependencySuccessors)) {
```

Comment:
Well, two things that showed up in the profiler here: 1) this method gets called _a lot_ (this is where most of the iterations through the task execution queue stop) so even something slight gets magnified and 2) this returns an ImmutableList, which is slightly slower to iterate than the original sets themselves.  This change did seem to move the needle a little bit - not by a lot, but enough to shave off a little time on a large task graph.